neurokit2/ecg/ecg_plot.py

```diff
@@ -138,40 +140,56 @@ def ecg_plot(ecg_signals, rpeaks=None, sampling_rate=None, show_type="default"):
         handles, labels = ax0.get_legend_handles_labels()
         order = [2, 0, 1, 3]
         ax0.legend(
-            [handles[idx] for idx in order], [labels[idx] for idx in order], loc="upper right"
+            [handles[idx] for idx in order],
+            [labels[idx] for idx in order],
+            loc="upper right",
         )
 
         # Plot heart rate.
         ax1.set_title("Heart Rate")
         ax1.set_ylabel("Beats per minute (bpm)")
 
-        ax1.plot(x_axis, ecg_signals["ECG_Rate"], color="#FF5722", label="Rate", linewidth=1.5)
+        ax1.plot(
+            x_axis,
+            ecg_signals["ECG_Rate"],
+            color="#FF5722",
+            label="Rate",
+            linewidth=1.5,
+        )
         rate_mean = ecg_signals["ECG_Rate"].mean()
         ax1.axhline(y=rate_mean, label="Mean", linestyle="--", color="#FF9800")
 
         ax1.legend(loc="upper right")
 
         # Plot individual heart beats.
-        if sampling_rate is not None:
-            ax2.set_title("Individual Heart Beats")
+        # TODO: how can we directly insert the figure from ecg_segment?
+        # fig_beats = ecg_segment(
+        #     ecg_signals["ECG_Clean"], peaks, sampling_rate, show="return"
+        # )
+        # ax2 = fig_beats.axes[0]
```

Comment:
Now `ecg_segment(..., show="return")` returns a Figure, but how can we insert that figure into the axis so that we don't have to re-plot it? 🤔  @danibene any ideas from the top of your head?

<sub>*matplotlib is the worse*</sub>

Reply:
Not off the top of my head, if I understood your question correctly it seems like it's not possible to insert axes from another figure? https://stackoverflow.com/questions/66369315/how-to-assign-axes-from-one-figure-to-axes-from-another-figure

Reply:
ok thanks, that was helpful, when with the `ax` kwarg similar to other packages